utils/ccdatofhir/mapping_resource_immunization.bal

```diff
@@ -0,0 +1,234 @@
+                xml<xml:Element> assignedEntityIdElement;
+                if isNamespaceAvailable {
+                    assignedEntityIdElement = assignedEntityElement/<v3:id>;
+                } else {
+                    assignedEntityIdElement = assignedEntityElement/<id>;
+                }
+
+                if assignedEntityIdElement is xml:Element && assignedEntityIdElement.length() > 0 {
+                    string|error assignedEntityId = assignedEntityIdElement.root;
+
+                    if assignedEntityId is string {
+                        r4:ImmunizationPerformer performer = {
+                            actor: {
+                                identifier: {
+                                    id: assignedEntityId
+                                }
+                            },
+                            'function: {
+                                coding: [
+                                    {
+                                        code: "AP"
+                                    }
+                                ]
+                            }
+                        };
+                        immunization.performer = [performer];
+                    } else {
+                        log:printDebug("assignedEntityId is not available", assignedEntityId);
+                    }
+                }
+            }
+        }
+        return immunization;
+    } else {
+        return;
+    }
+}
+
+isolated function mapCcdaStatusCodeToFhirImmunizationStatus(xml<xml:Element> codingElement) returns r4:ImmunizationStatus {
+    if codingElement is xml:Element {
+        string|error codeVal = codingElement.data();
+        if codeVal is string {
+            match codeVal {
+                "aborted" => {
+                    return r4:CODE_STATUS_NOT_DONE;
+                }
+                "cancelled" => {
+                    return r4:CODE_STATUS_NOT_DONE;
+                }
+                "completed" => {
+                    return r4:CODE_STATUS_COMPLETED;
+                }
+                "held" => {
+                    return r4:CODE_STATUS_NOT_DONE;
+                }
+                "new" => {
+                    return r4:CODE_STATUS_NOT_DONE;
+                }
+                "nullified" => {
+                    return r4:CODE_STATUS_ENTERED_IN_ERROR;
+                }
+                "obsolete" => {
+                    return r4:CODE_STATUS_NOT_DONE;
+                }
+                "suspended" => {
+                    return r4:CODE_STATUS_NOT_DONE;
+                }
+            }
+        } else {
+            log:printDebug("code is not available", codeVal);
+        }
+    }
+    return r4:CODE_STATUS_NOT_DONE;
+}
+
+isolated function mapCcdaToFhirImmunizationSite(xml<xml:Element> codingElement) returns string|error {
+    if codingElement is xml:Element {
+        string|error codeVal = codingElement.data();
```

Comment:
This function signature returns an error, but it does not return an error. So, shouldn't we handle this error?

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/817dadb5178bde7d809333a26f465c6688d3879a